Sources/MessagingPush/PushHandling/NotificationCenterDelegate.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import UserNotifications
+
+protocol PushEventListener {
+    var delegate: UNUserNotificationCenterDelegate { get }
+
+    func onPushClicked(_ push: PushNotification, completionHandler: @escaping () -> Void)
+    func shouldDisplayPushAppInForeground(_ push: PushNotification, completionHandler: @escaping (UNNotificationPresentationOptions) -> Void)
+
+    func newNotificationCenterDelegateSet(_ newDelegate: UNUserNotificationCenterDelegate?)
+    func beginListening()
+}
+
+struct PushNotification {
+    let pushId: String
+    let deliveryDate: Date
+    let title: String
+    let message: String
+    let data: [AnyHashable: Any]
+
+    init(notification: UNNotification) {
+        self.pushId = notification.request.identifier
+        self.deliveryDate = notification.date
+        self.title = notification.request.content.title
+        self.message = notification.request.content.body
+        self.data = notification.request.content.userInfo
+    }
+}
+
+class BaseSdkPushEventListener: NSObject, UNUserNotificationCenterDelegate, PushEventListener {
+    var delegate: UNUserNotificationCenterDelegate {
+        self
+    }
+
+    func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
+        onPushClicked(PushNotification(notification: response.notification), completionHandler: completionHandler)
+    }
```

Comment:
This is the main focus on this fix. 

Because we cannot directly call `func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {` in our tests, we instead...
1. have our tests call `onPushClicked(PushNotification(notification: response.notification), completionHandler: completionHandler)` 
2. in production, when iOS performs a call to `func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {`, it will simply make a request to `onPushClicked(PushNotification(notification: response.notification), completionHandler: completionHandler)`, which is a function that we have covered in tests.